From e5a94d955cd0dbef8b655d760456ef6de50b7b0d Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Tue, 27 Feb 2018 20:00:45 +0000
Subject: [PATCH 2/3] klte-common: Get rid of recovery-only fstab

 * Mounting /system partition is handled in kernel now,
   however removing the entry from fstab caused issues building
   the OTA. The workaround was to have a separate fstab, but turns out
   that simply setting the recoveryonly flag does the trick because
   those are then ignored during a normal Android boot.

Change-Id: I2944384d0a1c41bc9f9f51e2e29daff2bed0a0f4
---
 BoardConfigCommon.mk   |  2 +-
 rootdir/etc/fstab.full | 23 -----------------------
 rootdir/etc/fstab.qcom |  4 +---
 3 files changed, 2 insertions(+), 27 deletions(-)
 delete mode 100644 rootdir/etc/fstab.full

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index 4a23921..d54aae6 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -89,7 +89,7 @@ BOARD_HAS_LARGE_FILESYSTEM := true
 BOARD_HAS_NO_MISC_PARTITION := true
 BOARD_HAS_NO_SELECT_BUTTON := true
 BOARD_RECOVERY_SWIPE := true
-TARGET_RECOVERY_FSTAB := $(COMMON_PATH)/rootdir/etc/fstab.full
+TARGET_RECOVERY_FSTAB := $(COMMON_PATH)/rootdir/etc/fstab.qcom
 
 # SELinux
 include $(COMMON_PATH)/sepolicy/sepolicy.mk
diff --git a/rootdir/etc/fstab.full b/rootdir/etc/fstab.full
deleted file mode 100644
index 87bdf9c..0000000
--- a/rootdir/etc/fstab.full
+++ /dev/null
@@ -1,23 +0,0 @@
-# Android fstab file.
-# The filesystem that contains the filesystem e2fsck binary (typically /system) cannot
-# specify 'check', and must come before any filesystems that do specify 'check'
-
-#<src>                                              <mnt_point>     <type>  <mnt_flags and options>                                                                         <fs_mgr_flags>
-/dev/block/platform/msm_sdcc.1/by-name/system       /system         ext4    ro,errors=panic                                                                                 wait
-/dev/block/platform/msm_sdcc.1/by-name/userdata     /data           f2fs    nosuid,nodev,noatime,rw,inline_xattr                                                            wait,check,formattable,encryptable=footer,length=-16384
-/dev/block/platform/msm_sdcc.1/by-name/userdata     /data           ext4    nosuid,nodev,noatime,noauto_da_alloc,errors=continue,commit=20                                  wait,check,formattable,encryptable=footer,length=-16384
-/dev/block/platform/msm_sdcc.1/by-name/cache        /cache          f2fs    nosuid,nodev,noatime,rw,inline_xattr                                                            wait,check,formattable
-/dev/block/platform/msm_sdcc.1/by-name/cache        /cache          ext4    nosuid,nodev,noatime,noauto_da_alloc,errors=continue,commit=20                                  wait,check,formattable
-/dev/block/platform/msm_sdcc.1/by-name/apnhlos      /firmware       vfat    ro,shortname=lower,uid=1000,gid=1026,dmask=227,fmask=337,context=u:object_r:firmware_file:s0    wait
-/dev/block/platform/msm_sdcc.1/by-name/modem        /firmware-modem vfat    ro,shortname=lower,uid=1000,gid=1026,dmask=227,fmask=337,context=u:object_r:firmware_file:s0    wait
-/dev/block/platform/msm_sdcc.1/by-name/efs          /efs            ext4    nosuid,nodev,noatime,noauto_da_alloc,journal_async_commit,errors=panic                          wait,check
-/dev/block/platform/msm_sdcc.1/by-name/fota         /misc           emmc    defaults                                                                                        defaults
-
-/dev/block/platform/msm_sdcc.1/by-name/boot         /boot           emmc    defaults                                                                                        recoveryonly
-/dev/block/platform/msm_sdcc.1/by-name/recovery     /recovery       emmc    defaults                                                                                        recoveryonly
-
-/devices/msm_sdcc.3/mmc_host/mmc*                   auto            auto    defaults                                                                                        voldmanaged=sdcard1:auto,encryptable=userdata
-
-/dev/block/zram0                                    none            swap    defaults zramsize=25%,max_comp_streams=4,swapprio=10,notrim
-
-/devices/platform/xhci-hcd*                         auto            auto    defaults                                                                                        voldmanaged=usb:auto
diff --git a/rootdir/etc/fstab.qcom b/rootdir/etc/fstab.qcom
index fcf4f98..dd7bcf7 100644
--- a/rootdir/etc/fstab.qcom
+++ b/rootdir/etc/fstab.qcom
@@ -2,10 +2,8 @@
 # The filesystem that contains the filesystem e2fsck binary (typically /system) cannot
 # specify 'check', and must come before any filesystems that do specify 'check'
 
-# NOTE: /system partition is now early-mounted and the fstab entry is specified in device tree:
-#       /proc/device-tree/firmware/android/fstab/system
-
 #<src>                                              <mnt_point>     <type>  <mnt_flags and options>                                                                         <fs_mgr_flags>
+/dev/block/platform/msm_sdcc.1/by-name/system       /system         ext4    ro,errors=panic                                                                                 wait,recoveryonly
 /dev/block/platform/msm_sdcc.1/by-name/userdata     /data           f2fs    nosuid,nodev,noatime,rw,inline_xattr                                                            wait,check,formattable,encryptable=footer,length=-16384
 /dev/block/platform/msm_sdcc.1/by-name/userdata     /data           ext4    nosuid,nodev,noatime,noauto_da_alloc,errors=continue,commit=20                                  wait,check,formattable,encryptable=footer,length=-16384
 /dev/block/platform/msm_sdcc.1/by-name/cache        /cache          f2fs    nosuid,nodev,noatime,rw,inline_xattr                                                            wait,check,formattable
-- 
2.7.4

